src/main/java/com/epam/test_generator/services/IOService.java

```diff
@@ -41,8 +43,19 @@
      */
     public String generateFile(Long suitId, List<Long> caseIds) throws IOException {
         Suit suit = suitDAO.findById(suitId).orElseThrow(NotFoundException::new);
-        List<Case> cases = caseIds.stream()
-            .map(i -> caseDAO.findById(i).orElseThrow(NotFoundException::new))
+        List<Long> casesIdsFromSuit = suit.getCases().stream()
+            .map(Case::getId).collect(Collectors.toList());
+        if (!casesIdsFromSuit.containsAll(caseIds))
+            throw new NotFoundException();
+        List<Case> cases = suit.getCases().stream()
+            .filter(aCase -> caseIds.contains(aCase.getId()))
+            .sorted(Comparator.comparingLong(Case::getRowNumber))
+            .peek(aCase -> {
+                List<Step> orderedSteps = aCase.getSteps().stream()
+                    .sorted(Comparator.comparingLong(Step::getRowNumber))
+                    .collect(Collectors.toList());
+                aCase.setSteps(orderedSteps);
```

Comment:
IMO it would be better to do filtering without modifying hibernate entities. It can happen that these changes can be persisted.